lib/tasks/data/task.dart

```diff
@@ -22,5 +22,14 @@ class Task {
     }
   }
 
-  Task({this.title, this.subtasks, this.dueDate, this.priority, this.id});
+  Task({this.title, this.subtasks, this.dueDate, this.priority}) {
+    id = hashValues(title, subtasks, dueDate, priority);
+  }
+
+  void updateProperties(Task t) {
```

Comment:
This function will not be necessary. Instead, I recommend using a cascade operator ( `task..title = ` ) in the appropriate places.